src/csv_to_json.py

```diff
@@ -0,0 +1,65 @@
+import csv
+import yaml
+import json
+from pathlib import Path
+
+class bcolors:
+    HEADER = '\033[95m'
+    OKBLUE = '\033[94m'
+    OKCYAN = '\033[96m'
+    OKGREEN = '\033[92m'
+    WARNING = '\033[93m'
+    FAIL = '\033[91m'
+    ENDC = '\033[0m'
+    BOLD = '\033[1m'
+    UNDERLINE = '\033[4m'
+
+def main():
+    input_dir = Path('csv')
+    output_dir = Path('json')
+    output_dir.mkdir(parents=True, exist_ok=True)
+    mapping_path = Path('src/config.yaml')
+
+    with open(mapping_path, 'r', encoding='utf-8') as f:
+        config = yaml.safe_load(f)
+
+    spreadsheet_entries = config['spreadsheet']
+    skip = []
+    for e in spreadsheet_entries:
+        if e['skip'] is True:
+            skip.append(e['name'])
+
+    mapping_list = config['mapping']
+    mapping = {item['tess']: item['schema'] for item in mapping_list}
+
+    for csv_file in input_dir.glob('*.csv'):
+        name = str(csv_file).removeprefix('csv/').removesuffix('.csv')
+        if name in skip:
+            continue
+
+        with open(csv_file, 'r', encoding='utf-8') as f:
+            reader = csv.DictReader(f)
+            mapped_fields = [k for k in mapping if k in reader.fieldnames]
+            materials = []
+
+            for row in reader:
+                resource = {
+                    '@context': 'https://schema.org',
```

Comment:
```suggestion
                    '@context': 'http://schema.org',
```
It is still recommended to go for `http`, even though it will move at some point to `https`

Reply:
Good point!

Source https://schema.org/docs/developers.html:

> As noted in the [FAQ](https://schema.org/docs/faq.html#19), the Schema.org URI is sometimes written with 'http', and sometimes with 'https'. Consuming applications are encouraged to accept both representations, and we now generate downloads in both "flavours" as a convenience. In editorial work "on disk" we use the 'https' form. However other sites and applications are likely to be using 'http', especially in RDF representations. Although we encourage a move towards 'https', note in particular that the JSON-LD context file for Schema.org uses 'http', and we are unlikely to change this without a long warning period due to the impact it would have on deployed installations. Feedback on this is welcome, e.g. in this [github issue](https://github.com/schemaorg/schemaorg/issues/2886).